### PR TITLE
feat: Pin Post Feature 

### PR DESCRIPTION
**Pull request by developerfred**
- [x] Review attached mockup
 - [x] Ensure there can only be one pin per org
 - [x] Add Pin Post menu item in the dropdown after the message has been posted.
 - [x] Use Pin icon from Font Awesome
 - [x] Ensure only the org profile can pin the post and sees the control to pin / unpin
 - [x] Make sure a post can be unpinned
 - [x] Message the tribe chat channel(if ever one) when a post is pinned
 - [x] pin that post to the chat channel
 - [x] blast to the orgs chat townsquare
#6450

#### Screen
Screen Shot 2020-04-17 at 17 48 59

#### Feature Gif
pinned_task

#### Ensure only the org profile can pin the post and sees the control to pin / unpin
tribes-post

### CODE REVIEW

**Comment by androolloyd** [comment]
@PixelantDesign  :D

File: app/assets/v2/js/activity.js [hunk 1]
```diff
@@ -1045,6 +1045,37 @@ $(document).ready(function() {
...
+    if (state == 'unpin') {
+      $('.pinned-activity').addClass('bg-white');
+      $('.pinned-activity .tip_activity').css({'background-color': 'white'});
+      $('.pinned-activity').css({'border-bottom-color': '#EFEFEF;'});
+      $('.pinned-activity .activity_pinned').hide();
+      $('.box').removeClass('pinned-activity');
+      _alert('Sucess unpin.', 'success', 1000);
+    }
+
+    if (state == 'pinned') {
```
Comment (thelostone-mc) [inline app/assets/v2/js/activity.js:1063]: `else if`

File: app/retail/templates/shared/activity.html [hunk 2]
```diff
@@ -295,6 +295,15 @@
 
 
           <div class="dropdown-menu dropdown-menu-right shadow font-smaller-4 px-0">
+            {% if row.profile.organizations|length %}
+              <a class="pin_activity mr-2 dropdown-item px-3 font-smaller-5" data-toggle="tooltip" title="Pinned post will display at the top of the Tribe Town Square." href=# 
+              data-url={{row.url}}
+              data-affirmative=pinned
+              data-negative=unpin data-state="{% if row.metadata.pin%}pinned{% else %}unpin{%endif%}"
+              data-pk="{{row.pk}}">
+              <i class="fas fa-map-pin mr-2"></i>Pin Post
```
Comment (thelostone-mc) [inline app/retail/templates/shared/activity.html:304]: `<i class="fas fa-map-pin mr-2"></i> Pin Post`
Add an empty space

File: app/townsquare/models.py [hunk 3]
```diff
@@ -375,13 +376,31 @@ def __str__(self):
...
     activity = models.ForeignKey('dashboard.Activity', on_delete=models.CASCADE)
     created = models.DateTimeField(auto_now=True)
 
     def __str__(self):
         return f"Favorite {self.activity.activity_type}:{self.activity_id} by {self.user}"
 
+class Pin(SuperModel):
+    """Model for each Pinned."""
+
+    user = models.ForeignKey(get_user_model(), on_delete=models.CASCADE, related_name='pins')
```
Comment (androolloyd) [inline app/townsquare/models.py:390]: thinking perhaps we want to cxhange how the pin's are associated. 

Unlike favourites which are a user one to many, pin's are a one to one for to the target activity feed, perhaps excluding any filters user based filters. we could take a simple hash of the target feed and that becomes our unique basis for the pin. that way we can pin for the main feed, any specific tribe, hackathon or the like with little to no overhead.

Comment (PixelantDesign) [reply app/townsquare/models.py:390]: @developerfred could you take a look at @androolloyd comments?

**Comment by octavioamu** [comment]
Hey @developerfred seems the PR is mixing others commits not related to the task, can you  rebase and just keep your commits?

**Comment by octavioamu** [comment]
My recommendation is to cherrypick your commits into a new PR this one is mixing other people commits since the beginning, not sure what happened here.

File: app/townsquare/models.py (hunk 3 above)
Comment (developerfred) [reply app/townsquare/models.py:390]: Thanks @PixelantDesign and @androolloyd I will change this model.

Comment (androolloyd) [reply app/townsquare/models.py:390]: Fred ping me on chat and we can discuss the best way to create a unique key for each pin / activity feed.

Comment (developerfred) [reply app/townsquare/models.py:390]: Thanks @androolloyd I will do it now.

File: app/townsquare/models.py [hunk 4]
```diff
@@ -383,6 +383,29 @@ def __str__(self):
         return f"Favorite {self.activity.activity_type}:{self.activity_id} by {self.user}"
 
 
+class Pin(SuperModel):
+    """Model for each Pinned."""
```
Comment (owocki) [inline app/townsquare/models.py:387]: Pinned Post

Comment (developerfred) [reply app/townsquare/models.py:387]: @owocki i will change the name of the class to Pinned Post

**Comment by developerfred** [comment]
@danlipert https://github.com/gitcoinco/web/commit/fd1777254dcc398b291a789330b57f7e3bfa3205

**Comment by thelostone-mc** [comment]
@developerfred closing this out as https://github.com/gitcoinco/web/pull/6680 appends on to what you've built out here! Thanks for helping us get this done ^_^

**Pull request closed without merging**